File: python/house/house.py
"""Generate House verses."""

PREFIX = "This is"
SEGMENTS = [
    "house that Jack built.",
    "malt that lay in",
    "rat that ate",
    "cat that killed",
    "dog that worried",
    "cow with the crumpled horn that tossed",
    "maiden all forlorn that milked",
    "man all tattered and torn that kissed",
    "priest all shaven and shorn that married",
    "rooster that crowed in the morn that woke",
    "farmer sowing his corn that kept",
    "horse and the hound and the horn that belonged to",
]


def recite(start_verse: int, end_verse: int) -> list[str]:
    """Return a number of verses."""
    return [verse(i) for i in range(start_verse - 1, end_verse)]


def verse(verse_num: int) -> str:
    """Return one verse of the song."""
    parts = [SEGMENTS[i] for i in range(verse_num, -1, -1)]
    return " the ".join([PREFIX] + parts)

File: go/house/house.go
package house

import "strings"

var (
	prefix = "This is the "
	segments = []string{
		"horse and the hound and the horn\nthat belonged to",
		"farmer sowing his corn\nthat kept",
		"rooster that crowed in the morn\nthat woke",
		"priest all shaven and shorn\nthat married",
		"man all tattered and torn\nthat kissed",
		"maiden all forlorn\nthat milked",
		"cow with the crumpled horn\nthat tossed",
		"dog\nthat worried",
		"cat\nthat killed",
		"rat\nthat ate",
		"malt\nthat lay in",
		"house that Jack built.",
	}
)

func Verse(v int) string {
	parts := segments[len(segments) - v:]
	return prefix + strings.Join(parts, " the ")
}

func Song() string {
	parts := make([]string, len(segments))
	for i := 0; i < len(segments); i++ {
		parts[i] = Verse(i+1)
	}
	return strings.Join(parts, "\n\n")
}
